pkg/kubectl/cmd/cp.go

```diff
@@ -75,38 +75,41 @@ func NewCmdCp(f cmdutil.Factory, cmdOut, cmdErr io.Writer) *cobra.Command {
 }
 
 type fileSpec struct {
-	PodNamespace string
-	PodName      string
-	File         string
+	PodNamespace    string
+	PodName         string
+	File            string
+	LocalFileExists bool // indicates if File exists in local filesystem
 }
 
 var errFileSpecDoesntMatchFormat = errors.New("Filespec must match the canonical format: [[namespace/]pod:]file/path")
 
 func extractFileSpec(arg string) (fileSpec, error) {
-	pieces := strings.Split(arg, ":")
-	if len(pieces) == 1 {
-		return fileSpec{File: arg}, nil
-	}
-	if len(pieces) != 2 {
-		// FIXME Kubernetes can't copy files that contain a ':'
-		// character.
-		return fileSpec{}, errFileSpecDoesntMatchFormat
-	}
-	file := pieces[1]
-
-	pieces = strings.Split(pieces[0], "/")
-	if len(pieces) == 1 {
-		return fileSpec{
-			PodName: pieces[0],
-			File:    file,
-		}, nil
-	}
-	if len(pieces) == 2 {
-		return fileSpec{
-			PodNamespace: pieces[0],
-			PodName:      pieces[1],
-			File:         file,
-		}, nil
+
+	if pathExists(arg) {
+		return fileSpec{File: arg, LocalFileExists: true}, nil
+	} else {
+		if i := strings.Index(arg, ":"); i == -1 {
+			return fileSpec{File: arg}, nil
```

Comment:
return `fileSpec{File: arg}, errFileSpecDoesntMatchFormat`?

Reply:
this is not an error. in case `kubectl cp  tomcat-179433334-2f0b4:/tmp/a /tmp/c`, `/tmp/c`(dest) doesn't contain any colon and doesn't exist in local filesystem